### PR TITLE
Update helpers.py and storageclass.yaml of rbd csi

### PR DESCRIPTION
**Pull request by shreekarSS**
tests/helpers.py
  - Updated create_storage_class() with storageclass template update of rbd csi

templates/CSI/rbd/storageclass.yaml
 - Updated template storageclass.yaml as per issue #242 

Signed-off-by: Shreekar <sshreeka@redhat.com>

### CODE REVIEW

**Comment by shreekarSS** [comment]
test log:http://pastebin.test.redhat.com/773257
script execution is failed at the end of teardown due to ocs.exceptions.ResourceLeftoversException
PV was leftover after CephFS PVC deletion, bz https://bugzilla.redhat.com/show_bug.cgi?id=1722021

**Comment by shreekarSS** [comment]
> test log:http://pastebin.test.redhat.com/773257
> script execution is failed at the end of teardown due to ocs.exceptions.ResourceLeftoversException
> PV was leftover after CephFS PVC deletion, bz https://bugzilla.redhat.com/show_bug.cgi?id=1722021

purpose of test is passed

storageclass of RBD and CephFS
------------------------------------------
RBD

apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: storageclass-test-rbd-1916375474
  namespace: openshift-storage
parameters:
  adminid: admin
  clusterID: openshift-storage
  csi.storage.k8s.io/node-publish-secret-name: secret-test-rbd-1916375179
  csi.storage.k8s.io/node-publish-secret-namespace: openshift-storage
  csi.storage.k8s.io/provisioner-secret-name: secret-test-rbd-1916375179
  csi.storage.k8s.io/provisioner-secret-namespace: openshift-storage
  imageFeatures: layering
  imageFormat: '2'
  pool: cbp-test-1916373530
provisioner: rbd.csi.ceph.com
reclaimPolicy: Delete

-----------------------------------------
CephFS

apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: storageclass-test-cephfs-1916382042
  namespace: openshift-storage
parameters:
  csi.storage.k8s.io/node-stage-secret-name: secret-test-cephfs-1916380960
  csi.storage.k8s.io/node-stage-secret-namespace: openshift-storage
  csi.storage.k8s.io/provisioner-secret-name: secret-test-cephfs-1916380960
  csi.storage.k8s.io/provisioner-secret-namespace: openshift-storage
  monitors: rook-ceph-mon-a.openshift-storage.svc.cluster.local:6789,rook-ceph-mon-b.openshift-storage.svc.cluster.local:6789,rook-ceph-mon-c.openshift-storage.svc.cluster.local:6789
  pool: ocsci-cephfs-data0
  provisionVolume: 'true'
provisioner: cephfs.csi.ceph.com
reclaimPolicy: Delete

File: tests/helpers.py [hunk 1]
```diff
@@ -189,7 +189,11 @@ def create_storage_class(
     sc_data['parameters']['csi.storage.k8s.io/provisioner-secret-name'] = secret_name
     sc_data['parameters']['csi.storage.k8s.io/provisioner-secret-namespace'] = defaults.ROOK_CLUSTER_NAMESPACE
 
-    sc_data['parameters']['monitors'] = mons
+    if interface_type == constants.CEPHBLOCKPOOL:
+        sc_data['parameters']['clusterID'] = defaults.ROOK_CLUSTER_NAMESPACE
+    elif interface_type == constants.CEPHFILESYSTEM:
+        sc_data['parameters']['monitors'] = mons
```
Comment (vasukulkarni) [inline tests/helpers.py]: One is changing the clusterID and other the mons? Looks independent ?

Comment (shreekarSS) [reply tests/helpers.py]: yes as per upstream repo https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/csi/example/rbd/storageclass.yaml 
and 
https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/csi/example/cephfs/storageclass.yaml

**Comment by prsurve** [comment]
@shreekarSS need to update create_storageclass function from helpers.py to use this new param

**Comment by vasukulkarni** [comment]
@shreekarSS paste the logs please

**Comment by shreekarSS** [comment]
> @shreekarSS paste the logs please

test log:http://pastebin.test.redhat.com/773257

File: tests/helpers.py [hunk 2]
```diff
@@ -189,7 +189,11 @@ def create_storage_class(
     sc_data['parameters']['csi.storage.k8s.io/provisioner-secret-name'] = secret_name
     sc_data['parameters']['csi.storage.k8s.io/provisioner-secret-namespace'] = defaults.ROOK_CLUSTER_NAMESPACE
 
-    sc_data['parameters']['monitors'] = mons
+    if interface_type == constants.CEPHBLOCKPOOL:
+        sc_data['parameters']['clusterID'] = defaults.ROOK_CLUSTER_NAMESPACE
```
Comment (petr-balogh) [inline tests/helpers.py]: If this cluster namespace is going to be changed by at some point from custom config file then I would rather access this CLUSTER_NAMESPACE from config.ENV_DATA which refer all the time to one currently used for running cluster.  I won't block PR on this, but just be aware of this and maybe change in new PR if it suppose to reflect the cluster namespace of current cluster.

Comment (mbukatov) [reply tests/helpers.py]: +1

**Pull request merged**